src/v/iceberg/schema_avro.cc

```diff
@@ -200,8 +273,19 @@ struct_type_to_avro(const struct_type& type, std::string_view name) {
     const auto& fields = type.fields;
     for (const auto& field_ptr : fields) {
         const auto child_schema = field_to_avro(*field_ptr);
+
+        // Name holder is a stack-allocated placeholder which we can update and
+        // reference if the we need to sanitize the field name. Avoids an
+        // unnecessary copy of the name if the name is already valid.
+        ss::sstring name_holder;
+        auto& field_name = field_ptr->name;
+        if (!valid_avro_name(field_name)) {
+            name_holder = sanitize_avro_name(field_name);
+            field_name = name_holder;
+        }
+
         avro_schema.addField(
-          field_ptr->name,
```

Comment:
valid_avro_name(field->name) ? field->name : sanitize_avro_name(field->name),


instead of this song and dance with name_holder and the reference?

Reply:
oof I missed that. Yeah `std::string_view` for these local variables is your friend - I'm not sure what `addField` takes but I guess it has to be a `std::string_view`.